### PR TITLE
add test for checking direct return from linear combination of inputs

### PR DESCRIPTION
**Pull request by katat**
https://github.com/zksecurity/noname/issues/77

It passes the test. Here is the reasoning:
- the public var is always added as a single value var at the beginning.
- the circuit writer keeps these public vars as they are.
- so the public vars can be recovered to a single cell var from linear combination.
- the public output var and return var are two different vars (although there is an `assert_eq` constraint among them at the end).
- the return var can be a linear combination of multiple vars, which is not an issue as they don't need to be converted to a single cell var.

### CODE REVIEW

File: examples/fixture/asm/r1cs/lc_return.asm [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@ noname.0.7.0
+
+v_1 + v_2 == (v_3) * (1)
```
Comment (mimoo) [inline examples/fixture/asm/r1cs/lc_return.asm:3]: hopefully we'll see that with the reordering of public inputs it should be

```
v_1 + v_3 == v2
```

also what happened to write it like (A) * (B) = C ? it confuses me every time :D

**Review by mimoo — APPROVED** [review]
ohh, I guess the code I was looking at was about the public var, not the returned var!

File: examples/fixture/asm/r1cs/lc_return.asm (hunk 1 above)
Comment (katat) [reply examples/fixture/asm/r1cs/lc_return.asm:3]: currently, it is C = (A) * (B)

it is intended to align with the order of assignment in the noname code :D
so when it is in debug mode, this asm can align with the statement like
c = a + b

but this alignment might not be that useful though.

Comment (mimoo) [reply examples/fixture/asm/r1cs/lc_return.asm]: I guess it's a matter of taste :o doesn't really matter too much

**Pull request merged**